src/core/gui/XournalView.cpp

```diff
@@ -474,8 +477,10 @@ void XournalView::zoomChanged() {
     XojPageView* view = getViewFor(currentPage);
 
     ZoomControl* zoom = control->getZoomControl();
-    this->cache->setAnyZoomChangeCausesRecache(false);  // We're zooming -- no need for repeated re-renders.
-    this->cache->setRefreshThreshold(control->getSettings()->getPDFPageRerenderThreshold());
+    if (this->cache) {
+        this->cache->setAnyZoomChangeCausesRecache(false);  // We're zooming -- no need for repeated re-renders.
+        this->cache->setRefreshThreshold(control->getSettings()->getPDFPageRerenderThreshold());
+    }
```

Comment:
This should only happen when the settings are changed, see other comment.
This has a cost and the refresh threshold should just be updated upon settings change.
Is there a SettingsChanged event dispatch somewhere in the code?

(also the setAnyZoomChangeCausesRecache should be removed altogether, and the corresponding variable in PdfCache should be replaced by false all over)

Reply:
Ok. I found a way to only update those parameters when the settings are actually changed. Possibly not the best way, you tell me!